docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -314,6 +316,48 @@ This option is unnecessary for security for applications in production that only
 
 ---
 
+### Enable strict flexible arrays
+
+| Compiler Flag                                                         |      Supported since      | Description                                                                                   |
+|:--------------------------------------------------------------------- |:----------------------:|:--------------------------------------------------------------------------------------------- |
+| <span id="-fstrict-flex-arrays">`-fstrict-flex-arrays=3`</span>                             |       GCC 13<br/>Clang 16.0.0       | Make the compiler respect the sizes of trailing arrays more strictly (this affects bounds checking) |
+
+#### Synopsis
+
+Make the compiler respect the sizes of trailing arrays more strictly (this affects bounds checking).
+
+By default, GCC and Clang treat all trailing arrays (arrays that are placed as the last member or a structure) as flexible-sized arrays regardless of their actual size for the purposes of `__builtin_object_size()` calculations used by `__FORTIFY_SOURCE`. For example, given:
```

Comment:
Single leading underscore in `_FORTIFY_SOURCE`.

Reply:
Yikes! I don't know how that happened. I will fix that!